test/e2e/mcd_test.go

```diff
@@ -953,6 +953,9 @@ func TestMCDRotatesCertsOnPausedPool(t *testing.T) {
 	require.Nil(t, err)
 	t.Logf("Certificate stuck behind paused (as expected)")
 
+	// wait a few seconds for certs to rotate
```

Comment:
So, to be more precise, the issue here isn't that the certs take 5 seconds to rotate. The issue here is that in the MCO, we have a 5 second "cooldown" period between generating new rendered configs and propagating them to pools, hardcoded into the controller.

For testing this particular rotation, the issue is that there are actually 2 certs that get rotated, and depending on timing, those could show up as separate rendered configs, which now I think about it, shouldn't affect this test, but maybe that same "cooldown" period is being applied to the controllerconfig?

The 5 second "hack" I proposed is to (mostly) eliminate the possibility of this so the config is more settled. Fundamentally though, I was hoping we could fix it via either:
1. having a "smarter" way to detect whether the rotation has finished (and/or fix WaitForMCDToSyncCert)
2. having a retry in the next steps, allowing transient failures to settle
3. figure out where in the sync the delay is happening so we understand if this is an issue or not

I'm also fine with this fix as a temporary measure if we want to revisit this when we finish MCO-499, but I think it would be good to have extra comments on why this is being done so we don't forget the context

Reply:
Interesting, I was not aware of the 5 second cooldown - could you point me to where in the code this lives? I'd like to poke a bit more! I had assumed it was mostly arbitrary and we were just giving it more than "enough" time to complete the propagation. Definitely would like to do a permanent fix than something temporary! (:

Reply:
Sure! So we have these updateDelay's scattered throughout the code, e.g. controller https://github.com/openshift/machine-config-operator/blob/5843b7a4b27f15bc3313ce8a41ef6de792081fcb/pkg/controller/node/node_controller.go#L58 and https://github.com/openshift/machine-config-operator/blob/5843b7a4b27f15bc3313ce8a41ef6de792081fcb/pkg/daemon/daemon.go#L172

Although, now I look at the code again, I think that might have been a bit of a red herring in our case, since the controllerconfig sync should not have been gated by 5 seconds, nor the controllerconfig sync in the MCD. If I had to guess, maybe a 1-2 second sleep would have been enough as well?

The 5 second delay in the machineconfigs help us see the diff more clearly but I wonder where in the cert rotation path we are delayed